hg/utils.py

```diff
@@ -69,3 +73,37 @@ def copy_unique(model: ModelT) -> ModelT:
     if hasattr(copy, "uid"):
         setattr(copy, "uid", uid())
     return copy
+
+
+ViewInterval = Tuple[int, int]
+GenomicRegion = Tuple[str, int, int]
+
+
+def overlap(a: ViewInterval, b: ViewInterval) -> bool:
+    return a[1] >= b[0] and a[0] <= b[1]
+
+
+class Scale:
+    def __init__(self, chromsizes: Iterable[Tuple[str, int]]) -> None:
+        offsets = itertools.accumulate(map(itemgetter(1), chromsizes), initial=0)
```

Comment:
I wonder if the internal representation should just be a dataframe (`name`, `length`, `offset`), indexed by `name`?